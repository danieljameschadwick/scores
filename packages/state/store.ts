import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import themeReducer from "./reducer/ThemeReducer";

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, themeReducer);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [thunk],
});

export const persistor = persistStore(store);

export type AppState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  AppState,
  unknown,
  Action<string>>;

export default store;
