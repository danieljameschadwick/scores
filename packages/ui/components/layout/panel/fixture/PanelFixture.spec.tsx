import React from "react";
import { render, screen, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import store, { persistor } from "@scores/state/store";
import { PersistGate } from "redux-persist/integration/react";
import { PanelFixture } from "./PanelFixture";

const customRender = (id, home = {}, away = {}) => {
  return render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <PanelFixture id={id} home={home} away={away} />
      </PersistGate>
    </Provider>
  );
};

describe("PanelFixture component", () => {
  it("renders base loading component", async () => {
    const { queryByTestId } = customRender(1234);

    expect(queryByTestId("panel-fixture")).toBeTruthy();
    expect(queryByTestId("fixture-row-home")).toBeTruthy();
    expect(queryByTestId("fixture-row-away")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });

  // it("on click we navigate to fixture", async () => {
  //   const home = {};
  //   const away = {};
  //   const { queryByTestId } = customRender();

  //   fireEvent.click(queryByTestId("panel-fixture"));

  //   expect // trigger for router
  // });
});
