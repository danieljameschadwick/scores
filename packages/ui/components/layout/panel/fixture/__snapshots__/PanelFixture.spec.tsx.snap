// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelFixture component renders base loading component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 12,
      "width": "50%",
    }
  }
  testID="panel-fixture"
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "column",
          "paddingRight": "5%",
          "width": "70%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 2.5,
          },
        ]
      }
      testID="fixture-row-home"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 5,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          dataSet={
            Object {
              "media": undefined,
            }
          }
          style={
            Array [
              Array [
                Object {
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
          testID="name"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "width": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "700",
                  "textAlign": "center",
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
          testID="score"
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 2.5,
          },
        ]
      }
      testID="fixture-row-away"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginRight": 5,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": null,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          dataSet={
            Object {
              "media": undefined,
            }
          }
          style={
            Array [
              Array [
                Object {
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
          testID="name"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "width": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "700",
                  "textAlign": "center",
                },
              ],
              Object {
                "color": "#000",
              },
            ]
          }
          testID="score"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "justifyContent": "center",
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
        ]
      }
    >
      FT
    </Text>
  </View>
</View>
`;
